### ISSUE TITLE
[0] Change the NetworkPolicy Gatekeeper constraint for delete actions

### ISSUE THREAD

**Issue by linus-astrom** [elastisys/compliantkubernetes-apps#2073]
### Description

Change this [Gatekeeper constraint](https://github.com/elastisys/compliantkubernetes-apps/blob/main/helmfile.d/charts/gatekeeper/constraints/templates/networkpolicies/constraint.yaml) to no longer activate on delete actions.
As it could otherwise make the Deployments and other Resources be stuck in pending deletion if the associated NetworkPolicy gets deleted before the Deployment is or if both Resources get deleted at the same time depending on the finalizer.

### Definition of done

- [ ] The Gatekeeper constraint is changed.
- [ ] Test that Resources can be deleted without becoming stuck by this Gatekeeper constraint with `opa.networkPolicies.enforcement` set to `deny`.

**Comment by Zash** [comment]
_Only_ that particular constraint? Found a chart setting but I believe it applies widely.

https://github.com/elastisys/compliantkubernetes-apps/blob/7de50bb898d0bf647b9bad2e0bc02c77eb3785c3/helmfile.d/upstream/open-policy-agent-gatekeeper/gatekeeper/README.md?plain=1#L131

**Comment by linus-astrom** [comment]
> _Only_ that particular constraint? Found a chart setting but I believe it applies widely.
> 
> https://github.com/elastisys/compliantkubernetes-apps/blob/7de50bb898d0bf647b9bad2e0bc02c77eb3785c3/helmfile.d/upstream/open-policy-agent-gatekeeper/gatekeeper/README.md?plain=1#L131

Yes, only the specified webhook as its the one causing problems.
